tensorflow_probability/python/experimental/mcmc/with_reductions_test.py

```diff
@@ -0,0 +1,499 @@
+        new_sample, kernel_results])
+    self.assertEqual(kernel_results.streaming_calculations, 1)
+    self.assertEqual(new_sample, 1)
+    self.assertEqual(kernel_results.inner_results.counter_1, 1)
+    self.assertEqual(kernel_results.inner_results.counter_2, 2)
+
+  def test_boostrap_results(self):
+    fake_kernel = TestTransitionKernel()
+    fake_reducer = TestReducer()
+    reducer_kernel = WithReductions(
+        inner_kernel=fake_kernel,
+        reducers=fake_reducer,
+    )
+    pkr = self.evaluate(reducer_kernel.bootstrap_results(9.))
+    self.assertEqual(pkr.streaming_calculations, 0)
+    self.assertEqual(pkr.inner_results.counter_1, 0)
+    self.assertEqual(pkr.inner_results.counter_2, 0)
+
+  def test_is_calibrated(self):
+    fake_calibrated_kernel = TestTransitionKernel()
+    fake_uncalibrated_kernel = TestTransitionKernel(is_calibrated=False)
+    fake_reducer = TestReducer()
+    calibrated_reducer_kernel = WithReductions(
+        inner_kernel=fake_calibrated_kernel,
+        reducers=fake_reducer,
+    )
+    uncalibrated_reducer_kernel = WithReductions(
+        inner_kernel=fake_uncalibrated_kernel,
+        reducers=fake_reducer,
+    )
+    self.assertTrue(calibrated_reducer_kernel.is_calibrated)
+    self.assertFalse(uncalibrated_reducer_kernel.is_calibrated)
+
+  def test_tf_while(self):
+    fake_kernel = TestTransitionKernel()
+    fake_reducer = TestReducer()
+    reducer_kernel = WithReductions(
+        inner_kernel=fake_kernel,
+        reducers=fake_reducer,
+    )
+
+    initial_state = 0.
+    initial_kernel_results = reducer_kernel.bootstrap_results(
+        initial_state
+    )
+    def _loop_body(i, curr_state, pkr):
+      new_state, kernel_results = reducer_kernel.one_step(curr_state, pkr)
+      return (i + 1, new_state, kernel_results)
+
+    _, new_sample, kernel_results = tf.while_loop(
+        lambda i, _, __: i < 6,
+        _loop_body,
+        (0., initial_state, initial_kernel_results)
+    )
+
+    new_sample, kernel_results = self.evaluate([
+        new_sample, kernel_results])
+    self.assertEqual(kernel_results.streaming_calculations, 6)
+    self.assertEqual(new_sample, 6)
+    self.assertEqual(kernel_results.inner_results.counter_1, 6)
+    self.assertEqual(kernel_results.inner_results.counter_2, 12)
+
+  def test_nested_reducers(self):
+    fake_kernel = TestTransitionKernel()
+    nested_reducers = [[TestReducer(), TestReducer()], [TestReducer()]]
+    reducer_kernel = WithReductions(
+        inner_kernel=fake_kernel,
+        reducers=nested_reducers,
+    )
+    pkr = reducer_kernel.bootstrap_results(0.)
+    new_sample, kernel_results = reducer_kernel.one_step(0., pkr)
+    new_sample, kernel_results = self.evaluate([
+        new_sample, kernel_results])
+
+    self.assertEqual(
+        len(kernel_results.streaming_calculations[0]), 2)
+    self.assertEqual(
+        len(kernel_results.streaming_calculations[1]), 1)
+    self.assertEqual(
```

Comment:
For assertEqual, it's our style to do assertEqual(expected, actual)

Reply:
Oh man, I'm going to have to adjust a whole lot of tests haha

Reply:
Feel free to save it for later.

Reply:
probably best to do now and get it out of the way :') Thanks for letting me know! At least all future tests I write will be style appropriate

Reply:
also curious if there's similar style for other asserts (i.e. assertAllClose, assertNear) where there's distinctly one expected and one actual answer? I made the edits assuming there are for this PR but still good to know for the future!

Reply:
Yeah, it's followed for all equality type of asserts.